cwltool/workflow.py

```diff
@@ -593,6 +593,14 @@ def __init__(self,
         validation_errors = []
         self.tool = toolpath_object = copy.deepcopy(toolpath_object)
         bound = set()
+
+        if self.embedded_tool.get_requirement("SchemaDefRequirement")[0]:
+            if "requirements" not in toolpath_object:
+                toolpath_object["requirements"] = []
+            toolpath_object["requirements"].append(
+                self.embedded_tool.get_requirement("SchemaDefRequirement")[0]
```

Comment:
Interesting fix. I'm worried that this would allow references to custom types in the embedded tool without a matching workflow level import of the type; this would be contrary to the CWL v1.0.x spec.

Reply:
Ah yes, I didn't think about that. Do we actually need the new object we're creating to have the exact correct type, or could we just make the type `Any?`?

Reply:
On second look, this doesn't appear to have any side effects. Thanks!